### DISCUSSION TITLE
Allow Log level granularity for exception type in the Operator class

### DISCUSSION THREAD

**Discussion by BRMAXTED** [eadwinCode/django-ninja-extra#118] [Ideas]
In the Operation class, exceptions raised from controller functions are being logged as an error. 

> ERROR django       2024-03-30 00:54:16,524 operation 1 139945238042368 "POST - ProtectedUserController[create_user] /api/create" 403

It would be nice to modify the exception logging behavior to have granularity on the log level based on the exception.  I'd prefer to not have ERROR level logs generated from invalid client API request data for the given Request Schema endpoint.  

    def _prep_run(
        self, request: HttpRequest, temporal_response: HttpResponse, **kw: Any
    ) -> Iterator[RouteContext]:
        try:
            # Try code is here
        except Exception as e:
            self._log_action(
                request_logger.error,
                request=request,
                ex=e,
                extra={"request": request},
                exc_info=None,
            )
            raise e

**Comment by eadwinCode** [comment]
@BRMAXTED That's a good suggestion. But if I may ask, what type of `exception` are you raising in ProtectedUserController[create_user]? So that I can suggest a temporary solution while I look for a permanent fix. 

Also, your PR is welcomed if you have a solid approach to how this can be done.

**Comment by BRMAXTED** [comment]
@eadwinCode The particular exception that was logging an error was just pydantic ValidationError when the Request/Response data was not valid for the given Schema. A simple warning level log would suffice in that case.
